storage/tianmu/core/value_or_null.cpp

```diff
@@ -72,13 +73,20 @@ void ValueOrNull::GetBString(types::BString &tianmu_s) const {
 
 ValueOrNull::ValueOrNull(ValueOrNull const &von)
     : x(von.x),
-      sp(von.string_owner ? new char[von.len + 1] : von.sp),
       len(von.len),
       string_owner(von.string_owner),
       null(von.null) {
   if (string_owner) {
+    try {
+      sp = new char[len + 1];
+    } catch (const std::bad_alloc &e) {
```

Comment:
DO NOT catch this exception here, just throw it to upper.

Reply:
A system exception was thrown after the catch